### PR TITLE
Update README.md for new contributors

### LINKED ISSUE #97: Update README for new contributors
Currently, the README is a bit out of date:

- 'this github' -> this repository
- Make first two lines one line
- Move 'tested in Ubuntu...' into a list of tested architectures (Win, Linux, Mac, Mac arm)
- Remove api v1
- Add a diagram of all components and their interactions
- Update ML model description to match the new sparse tensor implementation
- Remove 'Where to add online updates' section
- Update Directory structure and Development workflow

### PR DESCRIPTION
**Pull request by sergia-ch**
Fix #97 

**View the file at https://github.com/tournesol-app/tournesol/tree/sergeivolodin-patch-1#readme**

image

Please check links, spelling, typos and for general sanity/completeness.

### CODE REVIEW

**Comment by sergeivolodin** [comment]
Right away: fix the diagram in dark mode (disable transparency)

image

**Comment by GresilleSiffle** [comment]
Great! the diagram is a very helpful addition : )

An idea: how about moving the contribution process into a `CONTRIBUTING.md` file?

After this pull request is merged, and if it's still revelant I'll glad to write (or contribute) to the how-to section with detailed instructions and commands to:
- fork tournesol/tournesol-app
- keep your fork synchronized with tournesol/tournesol-app
- work on a branch
  - edit your local source code
  - update the docker with your changes 
- run the linters / the tests
- creating pull request from your fork to tournesol/tournesol-app

Having a separate file for contribution instructions allow to add a lot of details, without worrying about the `README.md` size.

File: README.md [hunk 1]
```diff
@@ -1,218 +1,197 @@
...
+  <summary>⚙ <b>Setting up your development environment on your computer</b> (click to expand)</summary>
+  
+### Tested platforms  
+**Docker setup on Ubuntu, Debian, Windows 10, Mac OS X, Mac OS X with the M1 chip**
+    
+For expert use without Docker, the setup was tested on WSL 2 from Windows 10 Preview (no Docker), Debian GNU/Linux 10 (buster) (no Docker), Ubuntu 20.04 and 18.04 (no Docker). Mac OS X setup without Docker is more complicated, as an X server is required for integration tests, as well as some linux-specific dependencies.
 
+### Building the Docker image
 First, clone this repo and `cd` to it.
 You will need [Docker](https://docs.docker.com/get-docker/) installed and configured.
```
Comment (lfaucon) [inline README.md:32]: Can you say more about the "configured"? For me for example, I needed to increase the RAM (bump to 4G) and swap file size (bump to 2G) for everything to run smoothly

**Review by lfaucon — APPROVED** [review]
LGTM

**Pull request closed without merging**